libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -49,6 +49,13 @@ Declaration const* DeclarationContainer::conflictingDeclaration(
 			if (!dynamic_cast<FunctionDefinition const*>(declaration))
 				return declaration;
 	}
+	else if (dynamic_cast<EventDefinition const*>(&_declaration))
+	{
+		// check that all other declarations with the same name are events
```

Comment:
I currently don't see a general problem with allowing both functions and events with the same name, as long as the arguments are different. The point is that we need arguments for overload resolution, but that does not rule out mixing functions and events.